### PR TITLE
Propagate selector labels to timestamp collector

### PR DESCRIPTION
**Pull request by yulianedyalkova**
Towards https://github.com/giantswarm/giantswarm/issues/14653

And [this](https://github.com/giantswarm/giantswarm/issues/14653#issuecomment-745249794) is why we need this.

## Checklist

- [x] Update changelog in CHANGELOG.md.
- [ ] Update roadmap in ROADMAP.md.

### CODE REVIEW

File: pkg/controller/collector/set.go [hunk 1]
```diff
@@ -12,6 +14,7 @@ type SetConfig struct {
 	Logger               micrologger.Logger
 	K8sClient            k8sclient.Interface
 	NewRuntimeObjectFunc func() runtime.Object
+	Selector             selector.Selector
```
Comment (MarcelMue) [inline pkg/controller/collector/set.go:17]: This is basically a breaking change, or do I misunderstand?

Leaving this empty will cause no metrics to be collected AFAICT

Comment (yulianedyalkova) [reply pkg/controller/collector/set.go:17]: It's not. If you don't pass a selector to the controller it defaults to a selector that returns everything ([link](https://github.com/giantswarm/operatorkit/blob/cfe34c7b88e5d94355e15778f9e0545b93344144/pkg/controller/controller.go#L166-L168)) which will get propagated to the [controller set](https://github.com/giantswarm/operatorkit/blob/cfe34c7b88e5d94355e15778f9e0545b93344144/pkg/controller/controller.go#L185) too.

**Review by tfussell — APPROVED** [review]
LGTM.

**Review by fiunchinho — APPROVED** [review]
LGTM

**Review by MarcelMue — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,7 +11,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Added
 
-- Add `namespace` into controller setting. 
+- Add `namespace` into controller setting.
+- Propagate label selectors to timestamp collector
```
Comment (kopiczko) [inline CHANGELOG.md:15]: IMO that should go to  `### Fixed`.

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,7 +11,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Added
```
Comment (kopiczko) [inline CHANGELOG.md:12]:
```suggestion
### Added
```

**Review by kopiczko — APPROVED** [review]

**Pull request merged**